src/components/providers/authentication-provider.tsx

```diff
@@ -0,0 +1,78 @@
+import config from "../../app/config";
+import Keycloak from "keycloak-js";
+import { type Dispatch, type SetStateAction, useCallback, useEffect } from "react";
+import { type Auth } from "../../App";
+
+interface Props {
+  auth: Auth | undefined
+  setAuth: Dispatch<SetStateAction<Auth | undefined>>
+  userProfile: Keycloak.KeycloakProfile | undefined
+  setUserProfile: Dispatch<SetStateAction<Keycloak.KeycloakProfile | undefined>>
+  children: JSX.Element
+}
+
+const keycloak = new Keycloak(config.auth);
+/**
+* Provides Keycloak authentication functions, such as login and logout
+*/
+const AuthenticationProvider = ({ auth, setAuth, setUserProfile, children }: Props) => {
+  const updateAuthData = useCallback(() => {
+    console.log(auth);
+    setAuth({
+      token: keycloak.tokenParsed!,
+      tokenRaw: keycloak.token!,
```

Comment:
I think we should avoid Non-null Assertions (variables followed with ! operator). You could check these with an IF -statement beforehand.

Reply:
Fixed using Rome suggestion
https://docs.rome.tools/lint/rules/nononnullassertion/